### ISSUE TITLE
Validate fastify config objects

### ISSUE THREAD

**Issue by StarpTech** [fastify/fastify#793]
I think it's a very useful feature when the user receives immediate feedback when he uses a wrong configuration schema. It would also serve as a source of documentation. Hapi does it and it was a great feedback when experimenting with it for the first time. This could be easily solved with json-schemas.

We should also document all fastify options at a central place instead to scatter them on different pages it's very useful for a quick lookup.

```js
const fastify = require('../fastify')({
  logger: {
	logLevel: 'info' // not available
  }
}) // should throw
```
or
```js
const schema = {
  schemas: { // schema with `s`
    response: {
      200: {
        type: 'object',
        properties: {
          hello: {
            type: 'string'
          }
        }
      }
    }
  }
}

fastify
  .get('/', schema, function (req, reply) {
    reply
      .send({ hello: 'world' })
  }) // should throw
```
I think there are some more places.

**Comment by mcollina** [comment]
I think it would be a valuable feature. However plugins might need to extend those, and we might have to provide a way to ensure estensibility.

Would you like to give it a spin? Bear that it might be a disruptive change, and we might decide to park it (March 1st!).

**Comment by StarpTech** [comment]
Yes, I can work on it.

> However plugins might need to extend those, and we might have to provide a way to ensure estensibility.

This is no problem because we could enable "additionalProperties" at some places while validating the core options.

**Comment by jsumners** [comment]
Do you mean? https://github.com/fastify/fastify/blob/1ee7439d0a6fc1110db68c888f2fdcf5b370fdb5/docs/Factory.md

**Comment by StarpTech** [comment]
@jsumners never saw it. We should add it to the index and rename it to `Options`

**Comment by jsumners** [comment]
It's linked from https://github.com/fastify/fastify/blob/master/docs/Server-Methods.md#server-methods because

**Comment by delvedor** [comment]
Closed in #1471.
